server/src/main/java/io/crate/planner/operators/Rename.java

```diff
@@ -111,8 +111,9 @@ public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep)
         for (Symbol outputToKeep : outputsToKeep) {
             Symbols.intersection(outputToKeep, outputs, s -> {
                 Symbol childSymbol = parentToChildMap.get(s);
-                assert childSymbol != null : "There must be a mapping available for symbol " + s;
-                mappedToKeep.add(childSymbol);
+                if (childSymbol != null) {
```

Comment:
The reason why there is no mapping for the childSymbol is because there are two `(height > 0)` symbols with difference references.